### DISCUSSION TITLE
Multiple Smart Objects in One Mockup File

### DISCUSSION THREAD

**Discussion by bacmono** [joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script#16] [Q&A] [answered]
Hello, I've been using this script a lot lately, and find it truly useful. Thank You! 

One thing I'm struggling with now though, is making the script change multiple smart objects in one mockup file. More specifically, I'd like to have each input file of the batch changed in two placeholders 'at once'. 

I copied-pasted the _// Smart object #1_ in the code, renamed the _targets_ accordingly (I even assigned the two smart objects unique names in Ps), but with no success. See the code below. Either only one of the smart object is being changed, or none.
 
![PrtScr](https://user-images.githubusercontent.com/101568886/173821638-2feb6a97-ff37-4eb7-8494-9e93396dfded.png)

I appreciate any help.
Cheers..

**Comment by joonaspaakko** [comment, marked as answer]
What I might personally do in this situation is take [one of the examples](https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/tree/master/Examples) (in this repository) with multiple smart objects and replace the mockup with my own and see if that works better.

----

I don't see any obvious issues. Can you email me the project folder ...or at least `settings script.jsx, mockup.psd, and at least one input file`. It doesn't matter what the mockup or the input files look like, as long as it replicates the issue, in case it's not a free mockup.

**Comment by itorok88** [comment]
@joonaspaakko I have an issue with tha SmartObjects as well now. I have 2 mockup files with 6 smart objects each.
I've copied my working 3xSmartObject script and copied additional lines. But whatever I try, I only get 4 smartobjects filled.
Why does this happen?

Cheers!

**Reply by joonaspaakko** [reply]
My best guess is that there's some kinda disconnect between the settings file and the assets. Like an extra space or some other typo.

Just in case there's some kind of error in the script, I did [a test with 1 mockup with 13 smart objects](https://www.dropbox.com/sh/uokim17loigqyt1/AACysOj-woq2SEW1mowaW4tCa?dl=0) and I couldn't recreate the issue. I think I would have to look at your files to find out what's wrong. 

What you could try is edit the settings script to have just the problematic parts:

1. Remove one of the mockups (from the settings script)
2. Remove all working smart objects (from the settings script) — Maybe leave just one smart object that doesn't work...
3. Check the input folder path
4. Compare target name to layer name in the PSD mockup.
